tasks/task_manager.hh

```diff
@@ -95,6 +96,7 @@ public:
             foreign_task_vector _children;
             shared_promise<> _done;
             module_ptr _module;
+            abort_source _as;
```

Comment:
I recently found out that an `abort_source` on which no one called `subscribe()` generates an assert failure when abort is requested on it. See https://github.com/scylladb/seastar/blob/8ca9acda07893f2689c77ffea4114edbcab65d72/include/seastar/core/abort_source.hh#L109.
I don't see any subscribe calls here. Won't this cause problems?

Reply:
`_subscriptions` is of type `std::optional<subscription_list_type>` so the mentioned line checks whether an optional (not the list it contains) is empty.
By default it is set to an empty list (abort_source.hh:105) and exchanged with `std::nullopt` on the first call (abort_source.hh:111).
So I think that should not be a problem.

Reply:
> `_subscriptions` is of type `std::optional<subscription_list_type>` so the mentioned line checks whether an optional (not the list it contains) is empty. By default it is set to an empty list (abort_source.hh:105) and exchanged with `std::nullopt` on the first call (abort_source.hh:111). So I think that should not be a problem.

I agree, that's my understanding too

Reply:
You are right. I somehow didn't see the optional. I must have requested abort twice and somehow concluded that the problem is that I have no subscriptions.